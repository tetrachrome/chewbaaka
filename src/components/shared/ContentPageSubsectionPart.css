/**
 * ContentPageSubsectionPart.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 25, 2020
 * Updated  : Aug 07, 2020
 */

div.ContentPageSubsectionPartContainer {
  margin: 0 auto;
  padding: 20px 0px;
}
