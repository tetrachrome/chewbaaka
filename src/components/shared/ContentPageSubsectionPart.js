/**
 * ContentPageSubsectionPart.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 25, 2020
 * Updated  : Aug 07, 2020
 */

import React from 'react'

import './ContentPageSharedStyles.css'

import { getElementStyleClassName } from '../../styling/styling'

import './ContentPageSubsectionPart.css'

if ( process.env.NODE_ENV === 'development' )
  require('./ContentPageSubsectionPart-debug.css')

export default function ContentPageSubsectionPart({children}) {
  return (
    <div className={getElementStyleClassName("ContentPageSubsectionPartContainer")}>
      {children}
    </div>
  );
}
