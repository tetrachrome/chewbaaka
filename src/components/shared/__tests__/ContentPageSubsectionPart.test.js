/**
 * ContentPageSubsectionPart.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 26, 2020
 * Updated  : Jul 26, 2020
 */

import React from 'react'

import { render, screen } from '@testing-library/react'

import renderer from 'react-test-renderer'

import ContentPageSubsectionPart from '../ContentPageSubsectionPart'

const titleText = "Hello, World";

const bodyText = "This is some text.";

const renderChildren = () => (
  <div>
    <h1>{titleText}</h1>
    <p>{bodyText}</p>
  </div>
);

test('renders ContentPageSubsectionPart component', () => {
  render(
    <ContentPageSubsectionPart>
      {renderChildren()}
    </ContentPageSubsectionPart>
  );

  const titleElement = screen.getByText(titleText);
  expect(titleElement).toBeInTheDocument();

  const bodyTextElement = screen.getByText(bodyText);
  expect(bodyTextElement).toBeInTheDocument();
});

test('ContentPageSubsectionPart component snapshot', () => {
  const tree = renderer
    .create(
      <ContentPageSubsectionPart>
        {renderChildren()}
      </ContentPageSubsectionPart>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
