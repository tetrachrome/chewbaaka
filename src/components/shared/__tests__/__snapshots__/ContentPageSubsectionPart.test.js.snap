// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentPageSubsectionPart component snapshot 1`] = `
<div
  className="ContentPageSubsectionPartContainer"
>
  <div>
    <h1>
      Hello, World
    </h1>
    <p>
      This is some text.
    </p>
  </div>
</div>
`;
